scripts/replay-tx.js

```diff
@@ -1,31 +1,73 @@
+/**
+ * Message replayer script
+ * Requires a hash, network (string), and hash or block number
+ * If block number provided, it loops over the transaction history and replays all failed txs
+ */
 require("colors");
 const { Command } = require("commander");
 const log = require("single-line-log").stdout;
 const dotenv = require("dotenv").config();
 const { Contract, Wallet, utils } = require("ethers");
 const { getContractInterface } = require("@eth-optimism/contracts");
-const { JsonRpcProvider, getDefaultProvider } = require("@ethersproject/providers");
+const { JsonRpcProvider, InfuraProvider } = require("@ethersproject/providers");
 const program = new Command();
 
-program.requiredOption("-h, --hash <number>", "specify transaction hash");
 program.requiredOption("-n, --network <string>", "specify network");
-program.option("-k, --key <number>", "specify private key");
+program.requiredOption("-k, --key <number>", "specify private key");
+program.option("-h, --hash <number>", "specify transaction hash");
+program.option("-b, --blockNumber <number>", "specify a block number");
+
 program.parse(process.argv);
 const argOptions = program.opts();
 
-const mainnetProvider = getDefaultProvider(argOptions.network);
+if (argOptions.network !== "mainnet" && argOptions.network !== "kovan") {
+  console.error("Network must be 'mainnet' or 'kovan'");
+  process.exit();
+}
+
+if (!argOptions.hash && !argOptions.blockNumber) {
+  console.error("Must provide a hash or block number.");
+  return;
+}
+const provider = new InfuraProvider(argOptions.network, process.env.INFURA_KEY);
+
+// Addresses from May 11 regenesis
+const MAINNET_L1_MESSENGER_PROXY = "0x902e5fF5A99C4eC1C21bbab089fdabE32EF0A5DF";
+const KOVAN_L1_MESSENGER_PROXY = "0x78b88FD62FBdBf67b9C5C6528CF84E9d30BB28e0";
 
-const L1_MESSENGER_PROXY = "0xD1EC7d40CCd01EB7A305b94cBa8AB6D17f6a9eFE";
+const messengerAddress = argOptions.network === "mainnet" ? MAINNET_L1_MESSENGER_PROXY : KOVAN_L1_MESSENGER_PROXY;
 
-const wallet = new Wallet(argOptions.key, mainnetProvider);
-const proxyL1Messenger = new Contract(L1_MESSENGER_PROXY, getContractInterface("OVM_L1CrossDomainMessenger"), wallet);
+const wallet = new Wallet(argOptions.key, provider);
+const proxyL1Messenger = new Contract(messengerAddress, getContractInterface("OVM_L1CrossDomainMessenger"), wallet);
 
 const main = async () => {
-  const receipt = await mainnetProvider.getTransactionReceipt(argOptions.hash);
+  if (argOptions.hash) {
+    replayMessage(argOptions.hash);
+  } else {
+    // Replay all transactions from provided blockNumber
+    try {
+      const logs = await provider.getLogs({
+        address: messengerAddress,
+        topics: [utils.id(`RelayedMessage(bytes32)`)],
+        fromBlock: Number(argOptions.blockNumber,
+      });
```

Comment:
If this is connected to the L1 provider and L1 messenger, you should be looking for `SentMessage` events - replay all SentMessage (deposits) txs